domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/ValidFullAdmissionConstraintValidator.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.hmcts.cmc.domain.constraints;
+
+import uk.gov.hmcts.cmc.domain.models.response.FullAdmissionResponse;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import static uk.gov.hmcts.cmc.domain.constraints.utils.ConstraintsUtils.mayNotBeNullError;
+import static uk.gov.hmcts.cmc.domain.constraints.utils.ConstraintsUtils.mayNotBeProvidedError;
+import static uk.gov.hmcts.cmc.domain.constraints.utils.ConstraintsUtils.setValidationErrors;
+import static uk.gov.hmcts.cmc.domain.models.PaymentOption.FULL_BY_SPECIFIED_DATE;
+import static uk.gov.hmcts.cmc.domain.models.PaymentOption.IMMEDIATELY;
+import static uk.gov.hmcts.cmc.domain.models.PaymentOption.INSTALMENTS;
+
+public class ValidFullAdmissionConstraintValidator
+    implements ConstraintValidator<ValidFullAdmission, FullAdmissionResponse> {
+
+    public static class Fields {
+        static final String PAYMENT_DATE = "paymentDate";
+        static final String REPAYMENT_PLAN = "repaymentPlan";
+        static final String STATEMENT_OF_MEANS = "statementOfMeans";
+    }
+
+    @Override
+    public boolean isValid(FullAdmissionResponse fullAdmissionResponse, ConstraintValidatorContext context) {
+        switch (fullAdmissionResponse.getPaymentOption()) {
+            case IMMEDIATELY:
+                return validateImmediately(fullAdmissionResponse, context);
+            case FULL_BY_SPECIFIED_DATE:
+                return validateBySetDate(fullAdmissionResponse, context);
+            case INSTALMENTS:
+                return validateInstalments(fullAdmissionResponse, context);
+            default:
+                return false;
+        }
+    }
+
+    private boolean validateImmediately(FullAdmissionResponse value, ConstraintValidatorContext context) {
+        boolean valid = true;
+        String immediately = IMMEDIATELY.getDescription();
+
+        if (value.getPaymentDate().isPresent()) {
```

Comment:
Please remove, I will be sending payment date for immediate payment as it will get calculated as `today + 5 days`

Reply:
ok

Reply:
I have changed the logic. Now, `paymentDate` is mandatory.